testsuite/mailhog.py

```diff
@@ -47,7 +47,7 @@ def request(self, method: str = "GET", params: dict = None, endpoint: str = None
         """Requests the mailhog API"""
         params = params or {}
 
-        if not hasattr(self, "_session"):
+        if not self._session:
```

Comment:
I think this is an unnecessary change, I do not think this improves anything, and I still do not see a reason for this condition. However, I will leave this up to @mganisin if this change should be here or not

Reply:
More importantly these statements are not equal, this is change in behavior. If I remember this might be switch back to original statement. Comment (here in PR) may help to understand tihs.

Reply:
It checks if a session still exists during request method call. From time to time (during debugging happens that there is no session - None returned) should I remove it?